students/viveksyngh/quiz.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"encoding/csv"
+	"os"
+	"fmt"
+	"strings"
+	"log"
+	"flag"
+	"time"
+	"math/rand"
+) 
+
+type Question struct {
+	question string
+	answer string
+}
+
+func getQuestions(filePath string) ([]Question) {
+	file, err := os.Open(filePath)
+	if(err != nil){
+		log.Fatal("Failed to open file.")
+	}
+	reader := csv.NewReader(file)
+	questionList, err := reader.ReadAll()
+	if(err != nil) {
+		log.Fatal("Failed to parse CSV file.")
+	}
+	questions := make([]Question, 0)
```

Comment:
Now you have successfully loaded your `questionList`, you have the required length. You can do this instead:
```go
questions := make([]Question, len(questionList))
for i, question := range questionList {
	if len(question) != 2 {
		continue
	}
	questions[i] = Question{
		strings.TrimSpace(question[0]),
		strings.TrimSpace(question[1]),
	}
}
```
This way you don't allocate new array every time append is called. Also always check the input coming from outside world.